prog/dftb+/lib_dftb/dftd4param.F90

```diff
@@ -17,59 +17,67 @@ module dftbp_dftd4param
   use dftbp_dftd4refs
   implicit none
 
-  public :: TDftD4Calculator, TDispDftD4Inp, initializeCalculator
+  public :: TDftD4Calc, TDispDftD4Inp, TDftD4Calculator_init
+  public :: TDftD4Ref, TDftD4Ref_init, init
   public :: getEeqChi, getEeqGam, getEeqKcn, getEeqRad
   public :: getChemicalHardness, getEffectiveNuclearCharge, getSqrtZr4r2
   private
 
   !> Element-specific electronegativity for the electronegativity equilibration charges used in
-  !> DFT-D4.
+  !> DFT-D4
   interface getEeqChi
     module procedure getEeqChiSymbol
     module procedure getEeqChiNumber
   end interface getEeqChi
 
   !> Element-specific chemical hardnesses for the electronegativity equilibration charges used in
-  !> DFT-D4.
+  !> DFT-D4
   interface getEeqGam
     module procedure :: getEeqGamSymbol
     module procedure :: getEeqGamNumber
   end interface getEeqGam
 
   !> Element-specific CN scaling constant for the electronegativity equilibration charges used in
-  !> DFT-D4.
+  !> DFT-D4
   interface getEeqKcn
     module procedure :: getEeqKcnSymbol
     module procedure :: getEeqKcnNumber
   end interface getEeqKcn
 
-  !> Element-specific charge widths for the electronegativity equilibration charges used in DFT-D4.
+  !> Element-specific charge widths for the electronegativity equilibration charges used in DFT-D4
   interface getEeqRad
     module procedure :: getEeqRadSymbol
     module procedure :: getEeqRadNumber
   end interface getEeqRad
 
   !> Element-specific chemical hardnesses for the charge scaling function used to extrapolate the C6
-  !> coefficients in DFT-D4.
+  !> coefficients in DFT-D4
   interface getChemicalHardness
     module procedure :: getChemicalHardnessSymbol
     module procedure :: getChemicalHardnessNumber
   end interface getChemicalHardness
 
   !> Effective nuclear charges from the def2-ECPs used for calculating the reference
-  !> polarizibilities for DFT-D4.
+  !> polarizibilities for DFT-D4
   interface getEffectiveNuclearCharge
     module procedure :: getEffectiveNuclearChargeSymbol
     module procedure :: getEffectiveNuclearChargeNumber
   end interface getEffectiveNuclearCharge
 
-  !> PBE0/def2-QZVP atomic <r⁴>/<r²> expectation values.
+  !> PBE0/def2-QZVP atomic <r⁴>/<r²> expectation values
   interface getSqrtZr4r2
     module procedure :: getSqrtZr4r2Symbol
     module procedure :: getSqrtZr4r2Number
   end interface getSqrtZr4r2
 
 
+  interface init
+    module procedure :: TDftD4Calculator_init
+    module procedure :: TDftD4Ref_init
+  end interface
```

Comment:
I'd suggest not to overload, but use the explicit `T*_init()` routines when initializing the objects. (Usually, error messages due to argument mismatch on an overloaded routine are not very informative.)